packages/store-sync/src/recs/syncToRecs.ts

```diff
@@ -59,9 +66,49 @@ export async function syncToRecs<TConfig extends StoreConfig = StoreConfig>({
 
   world.registerDisposer(stopSync);
 
+  const functionSelectorToResourceSelector = new Map<Hex, Hex>();
+
+  const getResourceSelector = async (functionName: string): Promise<Hex> => {
+    const functionSignature = getAbiItem({
+      abi,
+      name: functionName,
+    }) as AbiFunction;
+
+    const worldFunctionSelector = getFunctionSelector(functionSignature);
+
+    if (functionSelectorToResourceSelector.has(worldFunctionSelector)) {
+      return functionSelectorToResourceSelector.get(worldFunctionSelector) as Hex;
+    }
+
+    const entity = encodeEntity(components.FunctionSelectors.metadata.keySchema, {
+      functionSelector: worldFunctionSelector,
+    });
+
+    let selectors = getComponentValue(components.FunctionSelectors, entity);
+
+    // If we can't find selectors due to not being synced yet, we can try to read them from the world contract
+    if (!selectors) {
+      const encodedValueSchema = valueSchemaToHex(components.FunctionSelectors.metadata.valueSchema);
+
+      const selectorRecord = (await publicClient.readContract({
+        address: address as Hex,
+        abi,
+        functionName: "getRecord",
+        args: [components.FunctionSelectors.id, [entity], encodedValueSchema],
```

Comment:
FYI if you rebase or merge main, the `valueSchema` arg here has changed to `fieldLayout`, which we'll need to compute from the schema (or store it in the component metadata when the table is registered)